packages/api/src/router/utils.ts

```diff
@@ -0,0 +1,44 @@
+import { eq } from "@openstatus/db";
+import { user, usersToWorkspaces, workspace } from "@openstatus/db/src/schema";
+
+import { Context } from "../trpc";
+
+/**
+ * Check if the user has access to the workspace, and return the workspace  and user  otherwise undefined
+ * @param  workspaceSlug
+ * @param  tenantId
+ * @param  ctx - trpc context
+ * @returns {Promise<{ workspaceId: string; userId: string }> | undefined} - workspaceId and userId
+ */
+export const hasUserAccessToWorkspace = async ({
+  workspaceSlug,
+  tenantId,
+  ctx,
+}: {
+  workspaceSlug: string;
+  tenantId: string;
+  ctx: Context;
```

Comment:
Here e.g. we can access tenantId by `ctx.auth.tenantId`, no need to pass it extra.